core/src/executor/select/mod.rs

```diff
@@ -154,6 +156,66 @@ fn into_rows(exprs_list: &[Vec<Expr>]) -> (Vec<Result<Row>>, Vec<String>) {
     (rows, labels)
 }
 
+fn sort_stateless(
```

Comment:
How about adding this to `executor/sort.rs` module.

Then, we can re-use partial sorting function.
e.g.
```rust
fn sort_by(values_a: &[(Value, Option<bool>)], values_b: &[(Value, Option<bool>)]) -> Ordering {
    let pairs = values_a
        .iter()
        .map(|(a, _)| a)
        .zip(values_b.iter())
        .map(|(a, (b, asc))| (a, b, asc.unwrap_or(true)));

    for (value_a, value_b, asc) in pairs {
        let apply_asc = |ord: Ordering| if asc { ord } else { ord.reverse() };

        match (value_a, value_b) {
            (Value::Null, Value::Null) => {}
            (Value::Null, _) => {
                return apply_asc(Ordering::Greater);
            }
            (_, Value::Null) => {
                return apply_asc(Ordering::Less);
            }
            _ => {}
        };

        match value_a.partial_cmp(value_b) {
            Some(ord) if ord != Ordering::Equal => {
                return apply_asc(ord);
            }
            _ => {}
        }
    }

    Ordering::Equal
}
```

Then, in `sort_stateless`
```rust
.sort_by(|(values_a, _), (values_b, _)| sort_by(values_a, values_b))
```
and in 'Sort::apply`
```rust
.sort_by(|(values_a, _, _), (values_b, _, _)| sort_by(values_a, values_b))
```

Then can share the same sorting function.